### ISSUE TITLE
DASH: Keep paused buffered data even if period is removed

### ISSUE THREAD

**Issue by ggabdol** [google/ExoPlayer#7081]
I'm playing dynamic DASH with multi-period. Is there a way to prevent ExoPlayer from jumping to the live edge while it's in paused state? The playback position jumps to the live-edge when MPD is updated.
Our expected behaviour is that when it's paused, the video screen remains the same even if the playback position gets out of the current period.
It seems that there are at least two places in the code that trigger the jumps. One is onInputFormatChanged() in MediaCodecRenderer, and the other one is https://github.com/google/ExoPlayer/blob/76962d50f1d80941d6768e4e765fa4ff010705e7/library/core/src/main/java/com/google/android/exoplayer2/ExoPlayerImplInternal.java#L503 in ExoPlayerImplInternal.
Is there an opinionated way to configure the player so it stops updating MPD when the player is paused and the playback position is out of the current period?
We're using ExoPlayer 2.9.4.

**Comment by ojw28** [comment]
> Is there a way to prevent ExoPlayer from jumping to the live edge while it's in paused state?

I don't think this is what ExoPlayer does. ExoPlayer will (or should) remain in the same playback position for as long as the player is paused. The seek bar will continue to update to indicate that more media is available, but the playback position itself will (or should) not change.

If this is not what you're seeing, please fill in the issue template with some concrete reproduction steps that would allow us to reproduce the issue.

**Comment by tonihei** [comment]
I think the reproduction steps above are already sufficient. 

The problem occurs because the current period is no longer part of the manifest and when the player handles the manifest update, it will realize that and seek to the default position of the next available period instead. 

This is in some way working as expected because there is no way playback can continue after the pause if the period doesn't even exist anymore. On the other hand, we allow pausing within a period even if the paused position falls behind the available live window. 

I'll mark this issue as an enhancement to better support this pausing feature, but I think it's relatively low priority given the implementation difficulties and the relatively niche benefit.
